src/main/java/study/step2/StringAddCalculator.java

```diff
@@ -0,0 +1,75 @@
+package study.step2;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import study.step2.exception.NumberConversionException;
+
+public class StringAddCalculator {
+    private static final int ZERO = 0;
+
+    private static final String DEFAULT_DELIMITERS = ",|:";
+    private static final String CUSTOM_DELIMITER_PATTERN = "//(.)\n(.*)";
+
+    private static final String NEGATIVE_INTEGER_EXCEPTION = "양수만 계산 가능합니다.";
+    private static final String NON_INTEGER_EXCEPTION = "%s 는 숫자가 아닙니다.";
+
+    private StringAddCalculator() {
+        throw new IllegalStateException("StringAddCalculator class");
+    }
+
+    public static int splitAndSum(String numberString) {
+        if (isNullOrEmptyString(numberString)) {
+            return ZERO;
+        }
+
+        List<String> numberByCustomDelimiter = splitByCustomDelimiter(numberString);
+        return numberByCustomDelimiter.isEmpty() ? add(splitNumberString(numberString, DEFAULT_DELIMITERS))
+                : add(numberByCustomDelimiter);
+    }
+
+    private static List<String> splitByCustomDelimiter(String numberString) {
+        Matcher matcher = getMatcher(CUSTOM_DELIMITER_PATTERN, numberString);
+        return matcher.find() ? splitNumberString(matcher.group(2), matcher.group(1)) : new ArrayList<>();
+    }
+
+    private static Matcher getMatcher(String pattern, String text) {
+        return Pattern.compile(pattern).matcher(text);
```

Comment:
Pattern을 생성하는 비용은 생각보다 비싼데요.
`DEFAULT_DELIMITERS`의 경우 변경되지 않기 때문에 재사용 할 수 있을 것 같습니다~

Reply:
regex 스트링을 상수화하기보단 Pattern 자체를 static 상수로 만들어서 재사용하도록 변경하겠습니다.
```suggestion
        private static final CUSTOM_DELIMITER_PATTERN = Pattern.compile("//(.)\n(.*)");
```